apps/3d_rec_framework/CMakeLists.txt

```diff
@@ -13,7 +13,7 @@ else(NOT VTK_FOUND)
 endif(NOT VTK_FOUND)
 
 # OpenNI found?
-if(NOT OPENNI_FOUND OR NOT BUILD_OPENNI)
+if(NOT OPENNI_FOUND OR NOT WITH_OPENNI)
```

Comment:
WITH_OPENNI | OPENNI_FOUND | NOT OPENNI_FOUND | NOT OPENNI_FOUND OR NOT WITH_OPENNI
------------|--------------|------------------|------------------------------------
F           | F            | T                | T
T           | F            | T                | T
T           | T            | F                | F

The last case doesn't happen. Btw. the `WITH_*` where introduced to deliberately disable searching for the dependency.